test/logger_test.rb

```
@@ -334,6 +341,25 @@
     end
   end
 
+  describe 'configuring error response' do
+    it 'blows up by default' do
+      assert_raises RuntimeError do
+        @logger.debug(message: true)
+      end
+    end
+
+    it 'silently swallows errors when configured to do so' do
+      apm = Minitest::Mock.new
+      apm.expect(:notify_error, nil, ["Logging schema validation error for {:message=>true}"])
+
+      @logger.configure_validation_error_response do |msg|
```

I think if we're going to assume a default of 'silently fail unless a validation error handling block is provided' then we'll need to make this very clear in the README, or there will be many tears.
I would instead advise a 'provide developer-focussed logging by default and make them turn it off for production' approach.